### PR TITLE
Added PlayerLevelChangeEvent

### PR DESCRIPTION
**Pull request by tudse145**
Created a new event which listens to when the player changes level

### CODE REVIEW

**Comment by luacs1998** [comment]
Why do you need this?

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 1]
```diff
@@ -413,4 +413,17 @@ public static void onPotionBrewed(ItemStack[] brewingItemStacks)
...
+    
+    public static int levelcheck(EntityPlayer player, int oldlevel, int newlevel)
+    {
+        if (oldlevel == -1) {
+            oldlevel = newlevel;
+        }   
+
+        if (oldlevel != newlevel) {
+            MinecraftForge.EVENT_BUS.post(new PlayerEvent.PlayerLevelChangeEvent(player, oldlevel, newlevel));
+            oldlevel = newlevel;
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java]: This does nothing

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 2]
```diff
@@ -413,4 +413,17 @@ public static void onPotionBrewed(ItemStack[] brewingItemStacks)
     {
         MinecraftForge.EVENT_BUS.post(new PotionBrewEvent.Post(brewingItemStacks));
     }
+    
+    public static int levelcheck(EntityPlayer player, int oldlevel, int newlevel)
+    {
+        if (oldlevel == -1) {
+            oldlevel = newlevel;
+        }
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java]: One line

File: src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java [hunk 3]
```diff
@@ -260,4 +260,18 @@ public File getPlayerFile(String suffix)
             return new File(this.playerDirectory, this.playerUUID+"."+suffix);
         }
     }
+    
+    public static class PlayerLevelChangeEvent extends PlayerEvent {
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java]: lack of javadocs

**Comment by LexManos** [comment]
Ya, why?

**Pull request closed without merging**